server/src/test/java/org/opensearch/index/seqno/SegmentReplicationLocalCheckpointTrackerTests.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.seqno;
+
+import org.junit.Before;
+import org.opensearch.test.OpenSearchTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class SegmentReplicationLocalCheckpointTrackerTests extends OpenSearchTestCase {
+
+    private LocalCheckpointTracker tracker;
+
+    public static LocalCheckpointTracker createEmptyTracker() {
+        return new LocalCheckpointTracker(SequenceNumbers.NO_OPS_PERFORMED, SequenceNumbers.NO_OPS_PERFORMED);
+    }
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        tracker = createEmptyTracker();
+    }
+
+    public void testSimpleSegrepPrimaryProcessed() {
```

Comment:
These should really be 3 separate tests since you're testing different use-cases

1. Base case
2. Idempotency
3. Persisted-vs-processed being out-of-sync, checkpoint update for seg-rep, and finally idempotency again